### ISSUE TITLE
Centroid calculation in 3D results in test failure

### ISSUE THREAD

**Issue by kks32** [cb-geo/mpm#106]
**Describe the bug**
When running `./mpmtest [cell]` results in failure of test on centroid calculation, while `./mpmtest [cell][3D]` runs fine, although the failure is in the 3D case when running for the `cell`.

**To Reproduce**
1. Compile ` cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_COMPILER=g++  ..; make -j8`
2. Run `./mpmtest [cell]`
3. Results in 
```
/home/krishna/research/mpm/mpm/tests/cell_test.cc:575: FAILED:
  REQUIRE( check_centroid[i] == Approx(centroid[i]).epsilon(Tolerance) )
with expansion:
  1.125 == Approx( 1.0 )
```

**Expected behavior**
In cell test in 3D
```
  REQUIRE( check_centroid[i] == Approx(centroid[i]).epsilon(Tolerance) )
```
should yield `1.0 == Approx( 1.0 )`

**Runtime environment (please complete the following information):**
 - OS/Docker image: Fedora 28
 - Branch: develop

**Additional context**
This happened after addition of `sides_indices` function to shapefn.

**Comment by kks32** [comment]
This is found to be an initialisation issue in the cell centroid calculation.

**Comment by kks32** [comment]
Fixed in #107
